plugins/extractors/metabase/client/client.go

```diff
@@ -61,9 +70,14 @@ func (c *client) GetTable(ctx context.Context, id int) (m.Table, error) {
 		return table, nil
 	}
 
+	const getTableRoute = "/api/table/{id}"
+	targetURL := c.urlb.New().
+		Path(getTableRoute).
+		PathParam("id", strconv.Itoa(id)).
+		URL()
+
 	var tbl m.Table
-	url := fmt.Sprintf("%s/api/table/%d", c.host, id)
-	if err := c.makeRequest(ctx, "GET", url, nil, &tbl); err != nil {
+	if err := c.makeRequest(ctx, getTableRoute, http.MethodGet, targetURL.String(), nil, &tbl); err != nil {
```

Comment:
What do you think of modifying this method to accept `urlbuilder.Source`? We can add a method on `urlbuilder.Source` to return the pattern that can be used along with the `URL()` method to construct the request and annotate it in the `makeRequest`.

Reply:
i did consider that too. but mostly didn't go with it to making too many refactors.